tracing-error/src/error.rs

```diff
@@ -101,31 +127,21 @@ where
         O: FnOnce(E) -> F,
         F: std::error::Error + Send + Sync + 'static,
     {
-        // # SAFETY
-        //
-        // This function + the repr(C) on the ErrorImpl make the type erasure throughout the rest
-        // of this struct's methods safe. This saves a function pointer that is parameterized on the Error type
-        // being stored inside the ErrorImpl. This lets the object_ref function safely cast a type
-        // erased `ErrorImpl` back to its original type, which is needed in order to forward our
-        // error/display/debug impls to the internal error type from the type erased error type.
-        //
-        // The repr(C) is necessary to ensure that the struct is layed out in the order we
-        // specified it, so that we can safely access the vtable and spantrace fields through a type
-        // erased pointer to the original object.
-        let vtable = &ErrorVTable {
-            object_ref: object_ref::<F>,
-        };
         let span_trace = self.inner.span_trace;
         let error = self.inner.error;
         let error = op(error);
 
-        TracedError {
-            inner: ErrorImpl {
-                vtable,
-                span_trace,
-                error,
-            },
-        }
+        TracedError::new(error, span_trace)
+    }
+
+    /// Convert the inner error type of a `TracedError` while preserving the
+    /// attached `SpanTrace` using the inner error's `Into` impl.
```

Comment:
an example would be nice here?

Reply:
(fwiw, when i said "an example would be nice", I was thinking just like, a small API usage example _here_, rather than a full-on `examples/` example...but that's also fine! :) )